benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/resources/BenchFlowExperimentResource.java

```diff
@@ -151,4 +152,21 @@ public BenchFlowExperimentStateResponse changeExperimentState(
 
     return new BenchFlowExperimentStateResponse(state);
   }
+
+  @POST
+  @Path("/abort")
+  public void abortExperiment(@PathParam("username") String username,
+      @PathParam("testName") String testName, @PathParam("testNumber") int testNumber,
+      @PathParam("experimentNumber") int experimentNumber) {
+
+    String experimentID =
+        BenchFlowConstants.getExperimentID(username, testName, testNumber, experimentNumber);
+
+    logger.info("request received: POST " + BenchFlowConstants.getPathFromExperimentID(experimentID)
+        + ABORT_PATH);
+
+    // execute in separate thread so we can return
```

Comment:
If it can take time, you might want to have an `ABORTING` state that is set just after receiving the request. Then probably you would also like to keep the last completed executed state.